docs/source/usage.rst

```diff
@@ -0,0 +1,230 @@
+   By default, it's accessed under *eav_values*:
+
+.. code-block:: python
+
+    patient.eav_values.all()
+    # = <QuerySet [has fever?: "True" (1), temperature: 37.7 (2)]>
+
+5. Sets *_eav_config_cls* attribute storing model of the config class
+   used by :class:`~eav.registry.Registry`. Defaults
+   to :class:`~eav.registry.EavConfig`; can be overridden (see below).
+
+With that out of the way, almost every aspect of the registration can
+be customized. All available options are provided to registration
+via config class: :class:`~eav.registry.EavConfig` passed to
+:meth:`~eav.register`. You can change them by overriding the class.
+Available options are as follows:
+
+1. ``manager_attr`` - Specifies manager name. Used to refer to the
+   manager from Entity class, "objects" by default.
+2. ``manager_only`` - Specifies whether signals and generic relation should
+   be setup for the registered model.
+3. ``eav_attr`` - Named of the Entity toolkit instance on the registered
+   model instance. "eav" by default. See attach_eav_attr.
+4. ``generic_relation_attr`` - Name of the GenericRelation to Value
+   objects. "eav_values" by default.
+5. ``generic_relation_related_name`` - Name of the related name for
+   GenericRelation from Entity to Value. None by default. Therefore,
+   if not overridden, it is not possible to query Values by Entities.
+
+Example registration may look like:
+
+.. code-block:: python
+
+    class SupplierEavConfig(EavConfig):
+        manager_attr = 'eav_objects'
+
+    eav.register(supplier, SupplierEavConfig)
+
+Additionally, :class:`~eav.registry.EavConfig` provides *classmethod*
+:meth:`~eav.registry.EavConfig.get_attributes` which is used to determine
+a set of attributes available to a given model. By default, it returns
+``Attribute.objects.all()``. As usual, it can be customized:
+
+.. code-block:: python
+
+    class SomeModelEavConfig(EavConfig):
+        @classmethod
+        def get_attributes(cls):
+            return Attribute.objects.filter(slug__startswith='a')
+
+Attribute validation includes checks against illegal attribute value
+assignments. This means that value assignments for attributes which are
+excluded for the model are treated with
+:class:`~eav.exceptions.IllegalAssignmentException`. For example (extending
+previous one):
+
+.. code-block:: python
+
+    some_model.eav.beard = True
+    some_model.save()
+
+will throw an exception.
+
+Creating Attributes
+-------------------
+
+Once your models are registered, you can starting creating attributes for
+them. Two most important attribute attributes (huh?) are *slug* and
+*datatype*. *slug* is a unique identifier and must be a valid Python
+variable name, as it's used to access values for that attribute from
+:class:`~eav.models.Entity` helper:
+
+.. code-block:: python
+
+    Attribute.objects.create(slug='color', datatype=Attribute.TYPE_TEXT)
+    flower.eav.color = 'red'
+
+    # Alternatively, assuming you're using default EntityManager:
```

Comment:
If that's *alternatively*, how are those two equal? Is `TYPE_TEXT` inferred from the supplied data for the attribute value? Is the attribute named "color" created on first use anywhere?